frontend/src/middleware/search.ts

```diff
@@ -0,0 +1,25 @@
+import { useSearchStore } from "~/stores/search"
+
+import type { Middleware } from "@nuxt/types"
+
+export const searchMiddleware: Middleware = ({ redirect, route, $pinia }) => {
+  const {
+    query: { q, searchBy },
+  } = route
+  /**
+   * This anonymous middleware redirects any search without a query to the homepage.
```

Comment:
What does "anonymous" mean here? Is that a Nuxt distinction?

Reply:
It _was_ anonymous before this change, and now that I've moved it to another file, the comment needs to be updated :) You can see in the changes that previously, `middleware` was a property of the `search.vue` page component.